### ISSUE TITLE
Declarative Repositories/Dependencies

### ISSUE THREAD

**Issue by seandenigris** [seandenigris/Simple-Persistence#9]
It seems we should be able to handle dependencies more declaratively. That is, instead of:
```smalltalk
restoreRepositories: someRepositories

	"My roots"
	Class1 restoreFrom: someRepositories first.
	Class2 restoreFrom: someRepositories second.
	Class3 restoreFrom: someRepositories fourth.
	
	"Dependencies"
	Project1DB restoreRepositories: someRepositories third.
	Project2DB restoreRepositories: someRepositories fourth.
```
we could have something like:
```smalltalk
clientClasses
	^ { Class1 .Class2. Class3 }.
```
and:
```smalltalk	
dependencies
	^ { Project1DB. Project2DB }
```

Related to #7

**Comment by seandenigris** [comment]
@gnaritas, thought you'd like to know that your SimplePersistence idea is still going strong! I had a dream for a while of making the API more declarative and finally got around to it with the last commit :)

**Comment by seandenigris** [comment]
Fixed by b807a1f.

**Comment by gnaritas** [comment]
Cool, glad to see my old code still inspiring.
